File: apps/ios/SofaExchange/Domain/Entities/City.swift
import Foundation

enum City: String, Codable, CaseIterable {
    case london    = "LONDON"
    case paris     = "PARIS"
    case milan     = "MILAN"
    case madrid    = "MADRID"
    case lisbon    = "LISBON"
    case berlin    = "BERLIN"
    case dublin    = "DUBLIN"
    case edinburgh = "EDINBURGH"
    case vienna    = "VIENNA"

    var displayName: String {
        NSLocalizedString("city_\(rawValue)", comment: "")
    }
}

File: apps/android/app/src/main/java/com/sofaexchange/domain/model/City.kt
package com.sofaexchange.domain.model

enum class City(val displayName: String) {
    LONDON("London"),
    PARIS("Paris"),
    MILAN("Milan"),
    MADRID("Madrid"),
    LISBON("Lisbon"),
    BERLIN("Berlin"),
    DUBLIN("Dublin"),
    EDINBURGH("Edinburgh"),
    VIENNA("Vienna");

    companion object {
        fun fromApiValue(value: String): City? = values().find { it.name == value }
    }
}
